compiler/src/dotty/tools/dotc/semanticdb/internal/SemanticdbInputStream.scala

```diff
@@ -448,6 +448,29 @@ class SemanticdbInputStream private (buffer: Array[Byte], input: InputStream) {
     }
   }
 
+  def readStringRequireUtf8(): String = {
+    val size: Int = readRawVarint32()
+    var bytes: Array[Byte] = Array()
+    var pos = bufferPos;
+    if (size <= (bufferSize - pos) && size > 0) {
+      // Fast path:  We already have the bytes in a contiguous buffer, so
+      //   just copy directly from it.
+      bytes = buffer;
+      bufferPos = pos + size;
+    } else if (size == 0) {
+      return "";
+    } else {
+      // Slow path:  Build a byte array first then copy it.
+      bytes = readRawBytesSlowPath(size);
+      pos = 0;
+    }
+    // TODO: should validate Utf8
```

Comment:
Where is this TODO coming from? Let's remove it if it's not needed.

Reply:
Removed!

Actually this method come from https://android.googlesource.com/platform/external/protobuf/+/7fca48d8ce97f7ba3ab8eea5c472f1ad3711762f/java/src/main/java/com/google/protobuf/CodedInputStream.java#408 and cemented out the validation part, because the callees of `readStringRequireUtf8` is only from test https://github.com/tanishiking/dotty/blob/6a534f298b4c3b85af4bd19dc039a36473f0382b/compiler/src/dotty/tools/dotc/semanticdb/Tools.scala#L47 and we can safely parse Semanticdb without validation.

Reply:
I'm not sure where they (the Scala codes under `semancidb.internal`  package) originally come from, if they are from https://android.googlesource.com/platform/external/protobuf/+/7fca48d8ce97f7ba3ab8eea5c472f1ad3711762f/java/src/main/java/com/google/protobuf , maybe we should add the copyright header 

```java
// Protocol Buffers - Google's data interchange format
// Copyright 2008 Google Inc.  All rights reserved.
// https://developers.google.com/protocol-buffers/
//
// Redistribution and use in source and binary forms, with or without
// modification, are permitted provided that the following conditions are
// met:
//
//     * Redistributions of source code must retain the above copyright
// notice, this list of conditions and the following disclaimer.
//     * Redistributions in binary form must reproduce the above
// copyright notice, this list of conditions and the following disclaimer
// in the documentation and/or other materials provided with the
// distribution.
//     * Neither the name of Google Inc. nor the names of its
// contributors may be used to endorse or promote products derived from
// this software without specific prior written permission.
//
// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

```

As it says,

> Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:

> //     * Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.